postgres/src/replication/state.rs

```diff
@@ -1,26 +1,106 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
 use sqlx::{PgPool, Type, postgres::types::Oid as SqlxTableId, prelude::FromRow};
+use tokio_postgres::types::PgLsn;
 
 use crate::schema::TableId;
 
 /// Table replication state as stored in the database
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum TableReplicationState {
```

Comment:
I have decided to copy the phase here, since from the perspective of this crate the phase doesn't exist and just the state is there.